### PR TITLE
Add Rails 7.2 support

### LINKED ISSUE #109: Rails 7.2 support
[Rails 7.2](https://rubygems.org/gems/rails) was released on August 9, 2024. Update rails-template to default to the 7.2 series.

Changelog: https://guides.rubyonrails.org/v7.2/7_2_release_notes.html
Upgrade guide: https://guides.rubyonrails.org/v7.2/upgrading_ruby_on_rails.html#upgrading-from-rails-7-1-to-rails-7-2

### PR DESCRIPTION
**Pull request by rahearn**
Changes:

* Generate rails 7.2 apps.
* general gem version updates
* add a dependabot config to the github actions generator
* small fix to `bin/trestle` and `bin/auditree` to pass CLI args into docker

closes #109

### CODE REVIEW

File: README.md
Comment (beechnut) [inline README.md:1]: change (non-blocking): Even though I think I was the person who wrote the readme, I was confused trying to follow it! The commands are nested inside numbers, then bullets, then sub-bullets, then parentheses. I 

change: It asked me about docker-trestle and compliance-as-code and I had no idea what to select and the readme doesn't explain it. Could you add something about that to guide a decision?

**Comment by rahearn** [comment]
Converting to draft as the `update` command needs more work.

**Review by beechnut — CHANGES_REQUESTED** [review]
I ran it, I have a project, and I ran `rails c` and `rails s` with the expected results.

However, the end of the log exited with a non-zero exit code? Not sure what's going on there.

```sh
# ... log log log ...
create mode 100644 zap.conf
         run  git add . from "."
         run  git commit -a -m 'Initial commit' from "."
On branch main
nothing to commit, working tree clean
⚠️   1   # <-- the non-zero exit
```

Otherwise, just left a couple suggested changes for the readme.

**Comment by rahearn** [comment]
> However, the end of the log exited with a non-zero exit code? Not sure what's going on there.

I'm 99% certain that's from the `git commit` resulting in "nothing to commit, working tree clean". Which is itself surprising, unless you ran it multiple times with the same destination directory?

**Comment by beechnut** [comment]
Just to double-check that, I ran `git status` in a repo with nothing to commit and a clean working tree and got

```
→  git status
On branch main
Your branch is up to date with 'origin/main'.

nothing to commit, working tree clean
🆗 # <-- this is my terminal's version of a 0 exit code
```

File: README.md [hunk 1]
```diff
@@ -49,19 +37,62 @@ There are a variety of options that customize your Rails application.
...
 _TODO: Documentation on whether you can override the `css` and `database` options._
 
+**Important:** Do not use flags `--skip-bundle` or `--skip-javascript`, or various parts of this template will break.
+
+#### Create your application
+
+<details><summary>If you are using Hotwire, run:</summary>
+
+```
+$ rails_template_18f new <project name> --hotwire ADDITIONAL_CONFIG_OPTIONS
```
Comment (rahearn) [inline README.md:64]: I don't love `ADDITIONAL_CONFIG_OPTIONS` but couldn't think of a better way to say "insert any --no-skip... options here"

**Review by pauldoomgov — APPROVED** [review]
Thanks for addressing the feedback! Approved!

**Pull request merged**